src/hooks/Guilds/ether-swr/guild/useProposalState.ts

```diff
@@ -0,0 +1,52 @@
+import { useTransactions } from 'contexts/Guilds/transactions';
+import { useERC20Guild } from 'hooks/Guilds/contracts/useContract';
+import moment from 'moment';
+import { useMemo } from 'react';
+import { useParams } from 'react-router-dom';
+import { useProposal } from './useProposal';
+
+enum ProposalState {
+  Active = 'Active',
+  Passed = 'Passed',
+  Executed = 'Executed',
+  Failed = 'Failed',
+}
+
+interface useProposalStateReturns {
+  data: {
+    isExecutable: boolean;
+    executeProposal: () => void;
+  };
+  error: Error;
+  loading: boolean;
+}
+
+function useProposalState(): useProposalStateReturns {
+  const { guild_id: guildId, proposal_id: proposalId } =
+    useParams<{ guild_id?: string; proposal_id?: string }>();

```

Comment:
I think we can actually remove the typings here as `useParams` from the router should resolve these properly.
Also the remapping we do on the line above can be fixed by updating the routing path.

We touched on these things in the call on Friday.

Reply:
so like this? ```  const { guild_id: guildId, proposal_id: proposalId } =
    useParams();```

Reply:
Lets ignore this here and pick it up on Monday during the refactor work (don't think its as simple as removing the typings)